### PR TITLE
Add OWL Client and IDO resource

### PR DESCRIPTION
**Pull request by cthoyt**
Tasks:

- [x] Extract shared functionality from OBO Client into superclass (d6a63dfd8)
- [x] Implement OWL Client class (d6a63dfd8)
- [x] Generate resources for IDO (e25dd2b5e and 195c2d6e2)
- [x] Update bioontology (b8cc9a866)
- [x] Fix issue in rebuilding bioontology (`python -m indra.ontology.bio build`) when adding hierarchy due to inconsistent entry in identifiers.org data (16f8bf223)

It turns out that the OWL Client and OBO Client were able to share most of the code for loading from disk, I just had to separate the implementations for getting the resources and building the caches. This also made updating the Bioontology very easy!

### CODE REVIEW

File: indra/databases/obo_client.py [hunk 1]
```diff
@@ -3,45 +3,40 @@
...
 __all__ = [
+    'OntologyClient',
     'OboClient',
     'RESOURCES',
 ]
 
-HERE = os.path.dirname(os.path.abspath(__file__))
-RESOURCES = os.path.join(HERE, os.pardir, 'resources')
+HERE = pathlib.Path(__file__).parent.resolve()
+RESOURCES = HERE.parent.joinpath('resources').resolve()
```
Comment (bgyori) [inline indra/databases/obo_client.py:21]: We actually have a function for getting paths to resource files now, see https://indra.readthedocs.io/en/latest/modules/resources.html, so if you're changing this bit of code anyway, might as well use those.

Comment (cthoyt) [reply indra/databases/obo_client.py]: Done in 06f333904

File: indra/resources/ido.json [hunk 2]
```diff
@@ -0,0 +1,1689 @@
+[
+ {
+  "id": "0000400",
+  "name": "symbiont role",
+  "namespace": "IDO",
+  "relations": {
+   "is_a": [
+    "IDO:0000401",
+    "IDO:0000402",
+    "IDO:0000403"
```
[the remark this answers is not in the document]
Comment (cthoyt) [reply indra/resources/ido.json:10]: what if there's a parent from a different namespace? Like [infectious disease course (IDO:0000495)](https://www.ebi.ac.uk/ols/ontologies/ido/terms?iri=http%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FIDO_0000495&viewMode=All&siblings=false) is a [disease course (OGMS:0000063)](https://www.ebi.ac.uk/ols/ontologies/ido/terms?iri=http%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FOGMS_0000063&viewMode=All&siblings=false). That shows up inside IDO.

[the remark this answers is not in the document]
Comment (bgyori) [reply indra/resources/ido.json:10]: I see, that's true - maybe the prefix removal should then be handled in the code somewhere.

[the remark this answers is not in the document]
Comment (cthoyt) [reply indra/resources/ido.json:10]: Also I just double checked that the other derived JSON files for GO, DOID, etc. have the same way of writing the full CURIEs

[the remark this answers is not in the document]
Comment (cthoyt) [reply indra/resources/ido.json:10]: The issue is stemming from the regex from identifiers.org being inconsistent with all other OBO ontologies. I'm going to override it in the code

[the remark this answers is not in the document]
Comment (cthoyt) [reply indra/resources/ido.json:10]: Fixed in 16f8bf223

File: indra/databases/obo_client.py [hunk 3]
```diff
@@ -70,6 +60,122 @@ def __init__(self, prefix, *, directory=RESOURCES):
         self.synonym_to_id = {k: v for k, v in self.synonym_to_id.items()
                               if k not in ambig_synonyms}
 
+    def count_xrefs(self):
```
Comment (cthoyt) [inline indra/databases/obo_client.py:63]: @bgyori this is not properly implemented and never used as far as I can tell. Can I remove it?

Comment (cthoyt) [reply indra/databases/obo_client.py:63]: *except in the `__main__`  of some derived clients. Maybe the implementation never quite made it to master

Comment (bgyori) [reply indra/databases/obo_client.py:67]: It looks like that's something you added originally, I don't think we have any usages so feel free to remove.

Comment (cthoyt) [reply indra/databases/obo_client.py:67]: lol

Comment (cthoyt) [reply indra/databases/obo_client.py]: Addressed in 21e431363

**Comment by cthoyt** [comment]
@bgyori failure on py36 seems to be coming from CTD code, unrelated.

**Comment by bgyori** [comment]
> @bgyori failure on py36 seems to be coming from CTD code, unrelated.

Yeah, I had to do another mock ontology build that solves that, thanks!

**Review by bgyori — APPROVED** [review]
This is great, thanks!

**Pull request merged**